service/src/main/java/com/epam/esm/service/service/user/UserServiceImpl.java

```diff
@@ -42,24 +49,26 @@ public UserServiceImpl(UserDAO dao,
 
     @Override
     @Transactional
-    public User create(User user) {
-        throw new UnsupportedOperationException();
+    public User create(User user) throws ValidationException, BadRequestException {
+        validator.validate(user);
+        try {
+            validateUserLogin(user.getLogin());
```

Comment:
please rename validateUserLogin as it's a bit confusing what is happening there as we already validated login in the previous method validator.validate(user);

and in this method you're checking if User already exists

Reply:
checkLoginIfVacant(String login)